src/corehost/cli/hostpolicy.cpp

```diff
@@ -443,16 +563,69 @@ SHARED_API int corehost_main_with_output_buffer(const int argc, const pal::char_
         else
         {
             trace::error(_X("Unknown command: %s"), g_init.host_command.c_str());
-            rc = LibHostUnknownCommand;
+            rc = StatusCode::LibHostUnknownCommand;
         }
     }
 
     return rc;
 }
 
+int corehost_delegate_init(const pal::string_t& location, arguments_t& args)
+{
+    if (trace::is_enabled())
+    {
+        trace_hostpolicy_entrypoint_invocation(location);
+        trace::info(_X("}"));
+
+        trace::info(_X("Deps file: %s"), g_init.deps_file.c_str());
+        for (const auto& probe : g_init.probe_paths)
+        {
+            trace::info(_X("Additional probe dir: %s"), probe.c_str());
+        }
+    }
+
+    // Host info should always be valid in the delegate scenario
+    assert(g_init.host_info.is_valid());
+
+    if (!parse_arguments(g_init, 0, nullptr, args))
+    {
+        return StatusCode::LibHostInvalidArgs;
+    }
+
+    args.trace();
+    return StatusCode::Success;
+}
+
+SHARED_API int corehost_get_coreclr_delegate(coreclr_delegate_type type, void **delegate)
+{
```

Comment:
Similar future proofing question - is this all the parameters we could imagine needing?

Reply:
See previous comment on `hostfxr_get_coreclr_delegate`.